src/Microsoft.IdentityModel.Tokens/Validation/Results/InternalTokenValidationResult.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.IdentityModel.Tokens
+{
+#nullable enable
+    /// <summary>
+    /// Internal class used to track the results of token validation and to provide a way to merge results.
+    /// Once all validation is complete, the results can be converted to a TokenValidationResult.
+    /// </summary>
+    internal class InternalTokenValidationResult
+    {
+        private bool _isValid;
+        private SecurityToken? _securityToken;
+        private TokenHandler _tokenHandler;
+        private List<ValidationResult> _validationResults = [];
+
+        /// <summary>
+        /// Creates a new instance of <see cref="InternalTokenValidationResult"/> to aggregate validation results.
+        /// </summary>
+        /// <param name="securityToken">The <see cref="SecurityToken"/> being validated.</param>
+        /// <param name="tokenHandler">The <see cref="TokenHandler"/> performing the validation.</param>
+        /// <exception cref="ArgumentNullException"></exception>
+        public InternalTokenValidationResult(SecurityToken? securityToken, TokenHandler tokenHandler)
+        {
+            _securityToken = securityToken;
```

Comment:
Why should securityToken be allowed to be null?

Reply:
Because there are cases such as validating a token from a string where the reading / decrypting can fail and we end up without a security token, and yet we need to return a result to indicate what went wrong.